src/main/java/marquez/api/resources/DatasourceResource.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.api.resources;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.CREATED;
+
+import com.codahale.metrics.annotation.ExceptionMetered;
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import java.util.List;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import marquez.api.exceptions.DatasourceUrnNotFoundException;
+import marquez.api.mappers.DatasourceResponseMapper;
+import marquez.api.models.DatasourceRequest;
+import marquez.common.models.ConnectionUrl;
+import marquez.common.models.DatasourceName;
+import marquez.common.models.DatasourceUrn;
+import marquez.service.DatasourceService;
+import marquez.service.exceptions.MarquezServiceException;
+import marquez.service.models.Datasource;
+
+@Slf4j
+@Path("/api/v1/datasources")
+public final class DatasourceResource {
+
+  private final DatasourceService datasourceService;
+
+  public DatasourceResource(@NonNull final DatasourceService datasourceService) {
+    this.datasourceService = datasourceService;
+  }
+
+  @POST
+  @ResponseMetered
+  @ExceptionMetered
+  @Timed
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  public Response create(@NonNull final DatasourceRequest datasourceRequest)
```

Comment:
@wslulciuc should we potentially call this `request` instead of `datasourceRequest`?

Reply:
Yep 👍

Reply:
You'll also need to annotate the request with [`@Valid`](https://docs.oracle.com/javaee/7/api/javax/validation/Valid.html)